### PR TITLE
AOCL spack recipes for 4.1 release w. new AOCL-Utils library

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
This adds version 4.1 of AOCL, including a new AOCL-Utils library.

License URLs have been updated, and various small tidyups have been made.

### CODE REVIEW

File: var/spack/repos/builtin/packages/aocl-sparse/package.py [hunk 1]
```diff
@@ -17,29 +19,24 @@ class AoclSparse(CMakePackage):
...
 
     _name = "aocl-sparse"
     homepage = "https://www.amd.com/en/developer/aocl/sparse.html"
     url = "https://github.com/amd/aocl-sparse/archive/3.0.tar.gz"
-    git = "https://github.com/amd/aocl-sparse.git"
+    git = "https://github.com/amd/aocl-sparse"
 
     maintainers("amd-toolchain-support")
 
+    version("4.1", sha256="68524e441fdc7bb923333b98151005bed39154d9f4b5e8310b5c37de1d69c2c3")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/aocl-sparse/package.py]: How is this checksum calculated?  

I'm getting `35ef437210bc25fdd802b462eaca830bfd928f962569b91b592f2866033ef2bb` from `spack checksum aocl-sparse`.

**Review by tldahlgren — COMMENTED** [review]
Confirmed the new homepage and version sha256s _except_ that of `aocl-sparse`.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Please correct or clarify the sha256.

File: var/spack/repos/builtin/packages/aocl-sparse/package.py (hunk 1 above)
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/aocl-sparse/package.py]: That checksum was apparently "calculated" by an error in fixing a conflict in our internal tree. Apologies

**Review by tldahlgren — DISMISSED** [review]
LGTM.  Did you want to review this PR as well @adamjstewart ?

**Comment by adamjstewart** [comment]
Nah, I don't use any of these packages. I'll trust your review.

**Comment by tldahlgren** [comment]
Looks like there are style issues with an extra blank line:

```
==> Running black checks
--- var/spack/repos/builtin/packages/amdblis/package.py	2023-08-22 19:28:33.866762 +0000
+++ var/spack/repos/builtin/packages/amdblis/package.py	2023-08-22 19:28:46.273936 +0000
@@ -43,11 +43,10 @@
     version("2.2", sha256="e1feb60ac919cf6d233c43c424f6a8a11eab2c62c2c6e3f2652c15ee9063c0c9")
 
     variant("ilp64", default=False, when="@3.0.1:", description="ILP64 support")
     variant("aocl_gemm", default=False, when="@4.1:", description="aocl_gemm support")
     variant("suphandling", default=True, description="Small Unpacked Kernel handling")
-
 
     def configure_args(self):
         spec = self.spec
         args = super().configure_args()
 
would reformat var/spack/repos/builtin/packages/amdblis/package.py
Oh no! 💥 💔 💥
1 file would be reformatted, 7 files would be left unchanged.
  black found errors
==> Running flake8 checks
var/spack/repos/builtin/packages/amdblis/package.py:50: [E303] too many blank lines (2)
  flake8 found errors
  ```

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**